ssh-pkcs11.c

```diff
@@ -819,6 +941,125 @@ pkcs11_fetch_ecdsa_pubkey(struct pkcs11_provider *p, CK_ULONG slotidx,
 }
 #endif /* HAVE_EC_KEY_METHOD_NEW */
 
+static int
+eddsa_validate_parameters(unsigned char *param, int len)
+{
+#ifdef NID_ED25519
+	const unsigned char *p = param;
+	ASN1_OBJECT *pObj = d2i_ASN1_OBJECT(NULL, &p, len);
+
+	/* Edwards EC public keys only support the use of the curveName
+	 * selection to specify a curve name as defined in [RFC 8032]
+	 * and the use of the oID selection to specify a curve through
+	 * an EdDSA algorithm as defined in [RFC 8410]. */
+	if (pObj != NULL) {
+		int nid = OBJ_obj2nid(pObj);
+		ASN1_OBJECT_free(pObj);
+		return (nid == NID_ED25519) ? (0) : (-1);
```

Comment:
Initially, after getting the PKCS#11 3.0 out, there was a bit unclear whether the parameters should be DER encoded OID or printableString "edwards25519". For that reason I was changing this in softhsm in the following PR:

https://github.com/opendnssec/SoftHSMv2/pull/526/files#diff-c2f98368371548f9c5c9ca522d545eeb756885689b8db595118c0ebe3a4c37c0R162-R191

I think the applications should ideally accept both of the versions to avoid unexpected failures. The above place is how I handled that in SoftHSM.

Reply:
Thanks! Implemented in a similar way, will be pushed soon